packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -257,84 +265,160 @@ export const buttonAppearanceVariantStyles = css``.withBehaviors(
-                    rgba(${actionRgbPartialColor}, 0.3) inset;
-            }
-
-            :host([appearance-variant='primary'][content-hidden]) .control {
-                padding: 0px;
+                background-color: ${buttonPrimaryBackgroundColor};
+                color: ${buttonPrimaryFontColor};
             }
 
             :host([appearance-variant='primary']) .control:hover {
-                border-color: ${borderHoverColor};
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+                background-color: transparent;
+                background-image: linear-gradient(
+                    ${buttonPrimaryBackgroundColor},
+                    ${buttonPrimaryBackgroundColor}
+                );
+                background-size: calc(100% - 4px) calc(100% - 4px);
+                color: ${buttonPrimaryFontColor};
             }
 
             :host([appearance-variant='primary']) .control${focusVisible} {
-                background-clip: border-box;
-                border-color: ${borderHoverColor};
-                border-width: ${borderWidth};
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-                padding: 0 ${standardPadding};
+                background-color: transparent;
+                background-image: linear-gradient(
+                    ${buttonPrimaryBackgroundColor},
+                    ${buttonPrimaryBackgroundColor}
+                );
+                background-size: calc(100% - 4px) calc(100% - 4px);
             }
 
-            :host([appearance-variant='primary'][content-hidden])
-                .control${focusVisible} {
-                padding: 0px;
+            :host([appearance-variant='primary']) .control:active {
+                background-image: linear-gradient(
+                    ${buttonPrimaryFillActionColor},
+                    ${buttonPrimaryFillActionColor}
+                );
+                background-size: calc(100% - 2px) calc(100% - 2px);
             }
 
-            :host([appearance-variant='primary']) .control:active {
-                box-shadow: none;
+            :host([appearance-variant='primary'][disabled]) .control {
+                background-color: rgba(${borderRgbPartialColor}, 0.1);
+                border-color: transparent;
+                color: rgba(${actionRgbPartialColor}, 0.3);
+                background-image: none;
             }
 
-            :host([appearance-variant='primary'][content-hidden])
-                .control:active {
-                padding: 0px;
+            :host([appearance-variant='primary']) [part='start'] {
+                ${iconColor.cssCustomProperty}: white;
             }
 
-            :host([appearance-variant='primary'][disabled]) .control {
-                background-clip: border-box;
+            :host([appearance-variant='primary'][disabled])
+                slot[name='start']::slotted(*) {
```

Comment:
I don't think all of these start and end slot selectors for targeting the icon color need `::slotted(*)`. Can you remove those to make it consistent?